src/utils.h

```diff
@@ -144,4 +144,16 @@ bool IsAbsolutePath(const std::string& path);
 bool IsUnixAbsolutePath(const std::string& path);
 bool IsWindowsAbsolutePath(const std::string& path);
 
+inline size_t hash_arguments(const std::vector<std::string>& args) {
```

Comment:
- rename to HashArguments
- move definition to cc file (or is there a pref reason for this to be inline?)